lib/ex_unit/lib/ex_unit/manifest.ex

```diff
@@ -0,0 +1,89 @@
+defmodule ExUnit.Manifest do
+  @moduledoc false
+
+  defmodule Entry do
```

Comment:
Let's use a record to mimic the behaviour in Mix manifests and let's make sure we version the manifest too. See the Elixir compiler manifest for some ideas: https://github.com/elixir-lang/elixir/blob/master/lib/mix/lib/mix/compilers/elixir.ex#L6

Reply:
What is the reason to prefer a record over a struct in this case?  I was under the impression that records are provided primarily to support working with erlang libraries, and that structs are generally superior.

Reply:
Because we are storing in disk, a record takes considerable less space, which speeds up loading and dumping time. We did some benchmarks at the time when building manifest comparing records, structs, serializing, ets, and others and "records + term serialization" was the fastest by far.